http/cluster-client/cluster_create.go

```diff
@@ -0,0 +1,91 @@
+	CloudProvider           string     `json:"cloud_provider" reform:"cluster_cloud"`
+	ClusterStatus           string     `json:"cluster_status" reform:"cluster_status"`
+	ClusterSize             int64      `json:"cluster_size" reform:"cluster_size"`
+	ClusterDbVendor         string     `json:"database_vendor" reform:"cluster_db_vendor"`
+	ClusterDbVersion        string     `json:"database_version" reform:"cluster_db_version"`
+	ClusterDbEndpoint       *string    `json:"database_endpoint" reform:"cluster_db_endpoint"`
+	ClusterInstanceSize     string     `json:"instance_size" reform:"cluster_instance_size"`
+	ClusterInstanceDiskType *string    `json:"cluster_instance_disk_type" reform:"cluster_instance_disk_type"`
+	ClusterInstanceIOPS     *uint64    `json:"iops" reform:"cluster_instance_iops"`
+	ClusterInstanceDiskSize *uint64    `json:"disk_size" reform:"cluster_instance_disk_size"`
+	EffectiveDiskSize       *uint64    `json:"effective_disk_size" reform:"effective_disk_size"`
+	HighAvailability        bool       `json:"high_availability" reform:"high_availability"`
+	CreatedAt               time.Time  `json:"created" reform:"created_at"`
+	UpdatedAt               time.Time  `json:"last_updated" reform:"updated_at"`
+	DeletedAt               *time.Time `json:"deleted_at" reform:"deleted_at"`
+	// DbAccount               DBAccount            `json:"database_account" reform:"db_account"`
+	Operable          bool    `json:"operable" reform:"-"`
+	NotOperableReason string  `json:"not_operable_reason" reform:"-"`
+	VpcUUID           *string `json:"vpc_uuid" reform:"vpc_uuid"`
+	SubnetUUID        *string `json:"subnet_uuid" reform:"subnet_uuid"`
+	// Tags                    Tags                 `json:"tags" reform:"tags"`
+	// AZS                     db.Strings           `json:"azs" reform:"azs"`
+	// MaintenanceSettings     *MaintenanceSettings `json:"maintenance_settings" reform:"maintenance_settings"`
+	// HostsInfo               HostsInfo            `json:"hosts_info" reform:"hosts_info"`
+	// CloudTags               CloudTags            `json:"cloud_tags" reform:"cloud_tags"`
+	UsePublicIPs bool   `json:"use_public_ips" reform:"use_public_ips"`
+	Notes        string `json:"notes" reform:"notes"`
+}
+
+// Create a new Clusters
+func (p Client) Create(ctx context.Context, c ccxprov.Cluster) (*ccxprov.Cluster, error) {
+	var b bytes.Buffer
+	if err := json.NewEncoder(&b).Encode(c); err != nil {
+		return nil, errors.Join(ccxprov.RequestEncodingErr, err)
+	}
+
+	url := p.connectionParams.BaseURL + "/api/prov/api/v2/cluster"
+	req, err := http.NewRequestWithContext(ctx, http.MethodPost, url, &b)
+	if err != nil {
+		return nil, errors.Join(ccxprov.RequestInitializationErr, err)
+	}
+
+	res, err := p.client.Do(req)
+	if err != nil {
+		return nil, errors.Join(ccxprov.RequestSendingErr, err)
+	}
+
+	if res.StatusCode != http.StatusCreated {
+		return nil, fmt.Errorf("%w: status = %d", ccxprov.ResponseStatusFailedErr, res.StatusCode)
```

Comment:
It's probably worth it to have an error message from the response body here as well.